src/Shared/HttpSys/RequestProcessing/NativeRequestContext.cs

```diff
@@ -760,9 +760,12 @@ private uint GetChunksHelper(ref int dataChunkIndex, ref uint dataChunkOffset, b
 
         var info = new Dictionary<int, ReadOnlyMemory<byte>>(count);
 
+        long fixup = (byte*)nativeRequest - (byte*)baseAddress;
+        var pRequestInfo = (HttpApiTypes.HTTP_REQUEST_INFO*)((byte*)nativeRequest->pRequestInfo + fixup);
```

Comment:
Is the `baseAddress` usage in `_backingBuffer.Memory.Slice((int)((long)requestInfo.pInfo - (long)baseAddress)` needed anymore if we're already adjusting the pointer?

Reply:
Comparing to https://github.com/dotnet/aspnetcore/blob/137c9aa6792390c20ed0f7c98468d4c9a7dda983/src/Shared/HttpSys/RequestProcessing/NativeRequestContext.cs#L812C79-L812C86, should that be + fixup instead?

Reply:
@BrennanConroy yes that other fixup is still needed because that is also a pointer to the possibly old location of that data. 

@Tratcher yes, good catch.

Reply:
> that is also a pointer to the possibly old location of that data

Ah, should have noticed the `p` prefix on the property 😄 I assumed it was the actual data so adding a pointer wouldn't have made sense.

Reply:
@BrennanConroy yes it's not confusing at all :) I knew there was a bug in this code and I started at it for quite a while and didn't catch it. It too a memory dump to find it.

Reply:
@Tratcher I think what I have is correct.